l10n_ch_lsv_dd/wizards/lsv_export_wizard.py

```diff
@@ -0,0 +1,485 @@
+
+    treatment_type = fields.Selection(
+        [('P', _('Production')), ('T', _('Test'))],
+        required=True,
+        default='P'
+    )
+    currency = fields.Selection(
+        [('CHF', 'CHF'),
+         ('EUR', 'EUR'),
+         ],
+        required=True,
+        default='CHF'
+    )
+    banking_export_ch_dd_id = fields.Many2one(
+        'banking.export.ch.dd',
+        'LSV file',
+        readonly=True
+    )
+    file = fields.Binary(
+        related='banking_export_ch_dd_id.file'
+    )
+    filename = fields.Char(
+        related='banking_export_ch_dd_id.filename',
+        size=256,
+        readonly=True
+    )
+    nb_transactions = fields.Integer(
+        'Number of Transactions',
+        related='banking_export_ch_dd_id.nb_transactions'
+    )
+    total_amount = fields.Float(
+        related='banking_export_ch_dd_id.total_amount'
+    )
+    state = fields.Selection(
+        [('create', _('Create')), ('finish', _('Finish'))],
+        readonly=True,
+        default='create'
+    )
+
+    @api.multi
+    def generate_lsv_file(self):
+        ''' Generate direct debit export object including the lsv file
+            content. Called by generate button.
+        '''
+        self.ensure_one()
+        payment_order_obj = self.env['account.payment.order']
+        payment_line_obj = self.env['bank.payment.line']
+
+        active_ids = self.env.context.get('active_ids', [])
+        if not active_ids:
+            raise exceptions.ValidationError(_('No payment order selected'))
+        payment_orders = payment_order_obj.browse(active_ids)
+
+        # common properties for all lines
+        properties = self._setup_properties(payment_orders[0])
+
+        total_amount = 0.0
+        lsv_lines = []
+
+        for payment_order in payment_orders:
+            total_amount = total_amount + payment_order.total_company_currency
+
+            ben_bank = payment_order.company_partner_bank_id
+            clean_acc_number = ben_bank.acc_number.replace(' ', '')
+            clean_acc_number = clean_acc_number.replace('-', '')
+            ben_address = self._get_account_address(ben_bank)
+            properties.update({
+                'ben_address': ben_address,
+                'ben_iban': clean_acc_number,
+                'ben_clearing':
+                    self._get_clearing(payment_order.company_partner_bank_id),
+            })
+
+            if not self._is_ch_li_iban(properties.get('ben_iban')):
+                raise exceptions.ValidationError(
+                    _('Ben IBAN is not a correct CH or LI IBAN (%s given)') %
+                    properties.get('ben_iban')
+                )
+
+            order_by = ''
+            if payment_order.date_prefered == 'due':
+                order_by = 'm.date_maturity ASC, '
+            order_by += 'al.partner_bank_id'
+
+            # A direct db query is used because order parameter in model.search
+            # doesn't support function fields
```

Comment:
order in `model.search` should be usable with stored computed fields. And by the way an SQL query will be as well impossible to be based on non stored fields.

Issue might be that you don't get a proper error as search will be silent about ordering by a non stored field.

Reply:
I did some corrections. However the code is not all mine, so I didn't change completely the SQL query. I just adapted it to the v10, because it was not working. I don't know if it's possible to use search with param such as oder='payment_line_ids.partner_bank_id' ? If it's ok for you I propose to keep this query (I removed the unused selected values).

Reply:
Ok we can keep it that way. But can you add a `# FIXME` to state that the current comment is not accurate?

Reply:
Done